Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessAllEffect.java

```diff
@@ -0,0 +1,58 @@
+
+package mage.abilities.effects.common.ruleModifying;
+
+import mage.abilities.Ability;
+import mage.abilities.Mode;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.constants.Duration;
+import mage.constants.Layer;
+import mage.constants.Outcome;
+import mage.constants.SubLayer;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+
+/**
+ * @author TheElk801
+ */
+public class CombatDamageByToughnessAllEffect extends ContinuousEffectImpl {
+
+    private final FilterCreaturePermanent filter;
+
+
+    public CombatDamageByToughnessAllEffect(FilterCreaturePermanent filter) {
+        this(filter, Duration.WhileOnBattlefield);
+    }
+     
+    public CombatDamageByToughnessAllEffect(FilterCreaturePermanent filter, Duration duration) {
+        super(duration, Layer.RulesEffects, SubLayer.NA, Outcome.Detriment);
+        this.filter = filter;
+    }
+    
+
+    private CombatDamageByToughnessAllEffect(final CombatDamageByToughnessAllEffect effect) {
+        super(effect);
+        this.filter = effect.filter;
+    }
+
+    @Override
+    public CombatDamageByToughnessAllEffect copy() {
+        return new CombatDamageByToughnessAllEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        game.getCombat().setUseToughnessForDamage(true);
+        game.getCombat().addUseToughnessForDamageFilter(filter);
+        return true;
+    }
+
+    @Override
+    public String getText(Mode mode) {
+        if (staticText != null && !staticText.isEmpty()) {
+            return staticText;
+        }
+        StringBuilder sb = new StringBuilder(filter.getMessage());
+        sb.append(" assigns combat damage equal to its toughness rather than its power");
+        return sb.toString();
```

Comment:
There's no reason to override `getText(Mode mode)` for this. Just set `staticText` in the constructor. Also, there's no reason to use a StringBuilder.